### PR TITLE
Add ArcFace Loss Implementation

### PR DESCRIPTION
**Pull request by aylinaydincs**
- This pull_request is a part of Google Summer of Code Project mentored bt @monatis at TensorFlow Organization.
- In this pull-request, I implement ArcFace Loss, and I also provide a sample notebook to show how to use it standalone and model.compile()
- I wrote a introductory medium story and a more comprehensive and advanced one will be coming soon
- I wrote unit tests for serializaiton and loss calculation

### CODE REVIEW

File: tensorflow_similarity/losses/arcface_loss.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+
+    def __init__(self,
+                 num_classes: int,
+                 embedding_size: int,
+                 margin: float = 0.50,  # margin in radians
+                 scale: float = 64.0,  # feature scale
+                 name: Optional[str] = "ArcFaceLoss",
+                 reduction: Callable = tf.keras.losses.Reduction.AUTO,
+                 **kwargs: object) -> object:
+        super().__init__(reduction=reduction, name=name, **kwargs)
```
Comment (aylinaydincs) [inline tensorflow_similarity/losses/arcface_loss.py:103]: I changed super().__init__(None, reduction=reduction, name=name, **kwargs) part and deleted the None. This needs to be solve the problem which occurs at testing pipeline

File: tensorflow_similarity/losses/arcface_loss.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+        m_hot = tf.one_hot(y_true, self.num_classes, on_value=self.margin, axis=1)
+        m_hot = tf.reshape(m_hot, [-1, self.num_classes])
+
+        cos_theta = tf.acos(cos_theta)
+        cos_theta += m_hot
+        cos_theta = tf.math.cos(cos_theta)
+        cos_theta = tf.math.multiply(cos_theta, self.scale)
+
+        cce = tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True, reduction=self.reduction)
+        loss: FloatTensor = cce(y_true, cos_theta)
```
Comment (aylinaydincs) [inline tensorflow_similarity/losses/arcface_loss.py:128]: At the same time, I change the type of the loss as FloatTensor, such that the other issue need to be solved as well.

**Comment by aylinaydincs** [comment]
change some lines of code according to previous logs

**Comment by owenvallis** [comment]
Looks like the current test are just failing the Black formatting checks for arcface_loss.py and the test file. Can you run Black and Isort on your local changes and reupload?
